tests/test_workers/test_tasks/test_utils.py

```diff
@@ -112,3 +115,43 @@ def test_different_pkg_managers(self, pkg_manager, assert_files_testdir):
 
         with pytest.raises(ValidationError, match=f"File check failed for {pkg_manager}"):
             af.present("absent_file")
+
+
+@pytest.mark.parametrize("id, state", [(2, "stale"), (3, "complete"), (1, "in-progress")])
+@mock.patch.object(requests_auth_session, "get")
+@mock.patch("cachito.workers.tasks.general.get_worker_config")
+def test_get_request_state(mock_config, mock_requests_get, id, state):
+    mock_requests_get.return_value.json.return_value = {"state": state}
+
+    config = mock_config.return_value
+    config.cachito_api_url = "http://cachito.domain.local/api/v1/"
+
+    assert get_request_state(id) == state
+    mock_requests_get.assert_called_once_with(f"http://cachito.domain.local/api/v1/requests/{id}")
+
+
+@pytest.mark.parametrize(
+    "id, state", [(1, "in_progress"), (2, "stale"), (3, "complete"), (None, "dummy")]
+)
+@mock.patch("cachito.workers.tasks.utils.get_request_state")
+def test_runs_if_request_in_progress(mock_get_state, id, state):
+    mock_get_state.return_value = state
+
+    @utils.runs_if_request_in_progress
+    def dummy_task(request_id):
+        return 42
+
+    if id is None:
+        with pytest.raises(
+            ValueError, match="Failed during state check: no request_id found for dummy_task task"
+        ):
+            assert dummy_task(id) is None
```

Comment:
```suggestion
            dummy_task(id)
```
The `assert` will not do anything here because `dummy_task(id)` will raise an exception. In general, assertions in a `pytest.raises` block will not work because the block is exited as soon as the exception is raised.

Which is fine, you don't have to assert anything here, only check that an exception was raised. To make that clear, you could remove the `assert` as suggested.